lightningd/options.c

```diff
@@ -379,7 +379,7 @@ static char *opt_set_hsm_password(struct lightningd *ld)
 	struct termios current_term, temp_term;
 	char *passwd = NULL;
 	size_t passwd_size = 0;
-	u8 salt[11] = "c-lightning";
+	u8 salt[16] = "c-lightning\0\0\0\0\0";
```

Comment:
Only a question :-) : why add these `\0` ?

Reply:
> Did valgrind catch this at all?

No..

> For reference the salt length is defined here

Yes, and I knew it : I don't know how I missed that in the first place..